components/script/script_thread.rs

```diff
@@ -2327,16 +2307,64 @@ impl ScriptThread {
     /// for the given pipeline (specifically the "navigate" algorithm).
     fn handle_navigate(&self, parent_pipeline_id: PipelineId,
                               browsing_context_id: Option<BrowsingContextId>,
-                              load_data: LoadData,
+                              mut load_data: LoadData,
                               replace: bool) {
         match browsing_context_id {
             Some(browsing_context_id) => {
                 let iframe = self.documents.borrow().find_iframe(parent_pipeline_id, browsing_context_id);
                 if let Some(iframe) = iframe {
                     iframe.navigate_or_reload_child_browsing_context(Some(load_data), NavigationType::Regular, replace);
                 }
+
+                // TODO: Test javascript: urls in iframes
             }
             None => {
+                let is_javascript = load_data.url.scheme() == "javascript";
+                if is_javascript {
+                    use url::percent_encoding::percent_decode;
```

Comment:
Move this to with the other use statements at the top of the file.

Reply:
Fixed in https://github.com/servo/servo/pull/17083/commits/5d28dd64d929ce471a7889a4bdddf8d1acb61184